### ISSUE TITLE
Full name for regular style should not require including “Regular”

### ISSUE THREAD

**Issue by khaledhosny** [googlefonts/fontbakery#853]
With Mada-Regular.ttf I get:
```
ERROR    [FULL_FONT_NAME:WINDOWS] entry: expected 'Mada-Regular' or 'Mada Regular' but got 'Mada'
```

But the spec says:
> Full font name; a combination of strings 1 and 2, or a similar human-readable variant. If string 2 is "Regular", it is sometimes omitted from name ID 4.

**Comment by davelab6** [comment]
Mac and windows  sections are different  and the error  doesn't make  this
clear enough

**Comment by khaledhosny** [comment]
The spec mentions no difference (it used to, but that was dropped years ago).

**Comment by davelab6** [comment]
For maximum utility Glyphs and I want to use the old style

**Comment by khaledhosny** [comment]
What applications/OS requires this, and what functionality will be lost?

**Comment by khaledhosny** [comment]
Anyway, you need to convince ufo2ft to do this, because UFO does not allow storing per-platform names. And FontForge as well.

**Comment by davelab6** [comment]
I'm happy to consider fontforge as obsolete and require its generated files to require hotfixing. 

I agree that ufo2ft needs to support this, so I've filed an issue there.

**Comment by davelab6** [comment]
> What applications/OS requires this, and what functionality will be lost?

I wonder if @schriftgestalt can comment

**Comment by anthrotype** [comment]
>  UFO does not allow storing per-platform names
you mean of course UFO2

I don't see where the problem is. The spec says

> If string 2 is "Regular", it is sometimes omitted from name ID 4

This is perfectly fine, and we've done this since ever.

**Comment by davelab6** [comment]
Oh I have assumed that the script is coded correctly. Let me check that.

**Comment by davelab6** [comment]
Okay yes, I confirmed Glyphs is exporting fonts in this way:

https://github.com/davelab6/glyphs-export/blob/567128d220141b3333edbff6eafd46499c1decdf/fonts/ttf/FamilySans-Regular._n_a_m_e.ttx#L60

**Comment by davelab6** [comment]
> `ERROR    [FULL_FONT_NAME:WINDOWS] entry: expected 'Mada-Regular' or 'Mada Regular' but got 'Mada'`

@felipesanches this is not right; I expect it to say,

`[FULL_FONT_NAME:WINDOWS] entry: expected 'Mada-Regular' but got 'Mada'`

`Mada Regular` is not appropriate for name records with id 4, as you can see in https://github.com/davelab6/glyphs-export/blob/567128d220141b3333edbff6eafd46499c1decdf/fonts/ttf/FamilySans-Regular._n_a_m_e.ttx#L18

**Comment by anthrotype** [comment]
That format looks like a PostScript name to me. The name ID 4 is "Full Name", it need not have any hyphen in it. The spec says it need to be "a combination of strings 1 and 2", and we have always simply concatenated the family and subfamily with a space (omitting Regular).

**Comment by davelab6** [comment]
> I don't see where the problem is. The spec says
>
> > If string 2 is "Regular", it is sometimes omitted from name ID 4
>
> This is perfectly fine, and we've done this since ever.

The problem is that I (and @m4rc1e who is even more keen on this than me) wants the collection to be uniform, and since we are using glyphsapp as the reference compiler, I want fonts to confirm to its output.

cc @graphicore

**Comment by anthrotype** [comment]
So basically you want ufo2ft to make nameID=4 ("Full Name") look the same as nameID=6 ("PostScript Name") but only on the Windows platform?

@khaledhosny you said the spec used to mention this? Have you got a link for that?

**Comment by davelab6** [comment]
> you want ufo2ft to make nameID=4 ("Full Name") look the same as nameID=6 ("PostScript Name") but only on the Windows platform?

I want it to match what Glyphs does.

**Comment by khaledhosny** [comment]
> >  UFO does not allow storing per-platform names
> you mean of course UFO2

Oh, I forgot about the support for them in UFO3. However, defcon does
not seem to support them, all I see is regular strings.

**Comment by anthrotype** [comment]
> defcon does not seem to support them, all I see is regular strings.

you can set `openTypeNameRecords` in `fontinfo.plist`:
https://github.com/typesupply/defcon/blob/ufo3/Lib/defcon/objects/info.py#L139

> I want it to match what Glyphs does. So, yes.

then maybe you need to make this clear in the ufo2ft issue.

**Comment by anthrotype** [comment]
@khaledhosny although ufo2ft currently does not read that key in fontinfo.plist (we shall open a separate issue/PR for that).

**Comment by khaledhosny** [comment]
> @khaledhosny you said the spec used to mention this? Have you got a link for that?

I can’t find any links to the old version on MS site, but the [change log](https://www.microsoft.com/typography/otspec/changes.htm) says name id 4 description was updated in version 1.9 in 2009, so here is a [link](https://web.archive.org/web/20080409142622/http://www.microsoft.com/typography/otspec/name.htm) from the wayback machine. Interestingly the exception is mentioned is for CFF fonts, so even if we were to follow the old spec, this should be done for fonts with CFF table only not the ones with glyf table (fontbakery does not even seem to support fonts with CFF table).

**Comment by davelab6** [comment]
> fontbakery does not even seem to support fonts with CFF table

No currently it is out of scope.

**Comment by davelab6** [comment]
Maybe the name table scheme that @schriftgestalt has devised is buggy?

**Comment by khaledhosny** [comment]
Copying the old wording here:
>  	Full font name; this should be a combination of strings 1 and 2. Exception: if the font is "Regular" as indicated in string 2, then use only the family name contained in string 1.
An exception to the above definition of Full font name is for Microsoft platform strings for CFF OpenType fonts: in this case, the Full font name string must be identical to the PostScript FontName in the CFF Name INDEX.

So the requirement was that CFF font name and name id 4 for MS platform be identical, so the check here is meaningless since there is no CFF table to begin with, and if were wre to support CFF in the future then that what you be comparing to. Note that even name id 6 is no longer required to match the CFF font name, since font collections can have multiple fonts sharing the same CFF table.

So in short I suggest forgetting about all this legacy mess, and ask Glyphs app to be fixed if it os producing such names in TTF fonts.

**Comment by schriftgestalt** [comment]
I had a look and it seem to be a mistake. It the windows style linking name is 'Regular', it would put the font name in name id 4 instead of the full name. I did that to overwrite makeOTF that would omit the 'Regular'. Now the Windows and Mac name id 4 will always have the 'Regular'.

**Comment by anthrotype** [comment]
It is perfectly OK to omit the 'Regular' as the spec says.

**Comment by schriftgestalt** [comment]
But I like the Regular. It gives a hole in some font menus. It is fine for four style families but with more members I find it odd if one entry in the list has a much shorter name.

**Comment by davelab6** [comment]
Aha! Thanks for checking on this, Georg.

**Comment by davelab6** [comment]
On Jul 26, 2016 2:46 PM, "Cosimo Lupo" <notifications@github.com> wrote:
>
> It is perfectly OK to omit the 'Regular' as the spec says.

I think it would be better to make the 2 sets of Mac/win records as similar
as possible

**Comment by anthrotype** [comment]
> better to make the 2 sets of Mac/win records as similar
as possible

Yeah, either both containing "Regular" or none. Exactly because it is just a preference, it could be left to the user to decide.

**Comment by schriftgestalt** [comment]
Both have Regular now.

**Comment by felipesanches** [comment]
I'm a bit confused about what should fontbakery be enforcind, then. @davelab6 Is the current implementation of the fontbakery check considered good and we can cose this issue or should I still change the implementation ?

**Comment by davelab6** [comment]
Yeah we need to change it to be like Glyphs. Eg, id4 is "Family Sans
Regular" for both platform ids

**Comment by khaledhosny** [comment]
This is still not fixed:
```
ERROR    [FULL_FONT_NAME:WINDOWS] entry: expected 'Mada-Medium' but got 'Mada Medium'
````

**Comment by felipesanches** [comment]
Would the good rule for NAMEID_FULL_FONT_NAME be this, then?

```python
if style == "Regular":
  expected_value = "{}-{}".format(fname, style)
else:
  expected_value = "{} {}".format(fname_with_spaces, style_with_spaces)
```

Or maybe it is some other naming rule?

**Comment by felipesanches** [comment]
I'm trying to accomodate @davelab6 naming rules requests as described in https://github.com/googlefonts/fontbakery/issues/853#issuecomment-235450285 and https://github.com/googlefonts/fontbakery/issues/853#issuecomment-235346229

**Comment by davelab6** [comment]
In the next release of Glyphs the Regular will not be different to the other styles, it will be `"{} {}".format(fname_with_spaces, style_with_spaces)`

**Comment by felipesanches** [comment]
@davelab6 OK, please review https://github.com/felipesanches/fontbakery/commit/179a6db194103a9891c3fad5f8d3cb6eb5511a51

**Comment by davelab6** [comment]
@felipesanches looks good
